lib/buffer.js

```diff
@@ -553,64 +564,149 @@ function base64ByteLength(str, bytes) {
   return (bytes * 3) >>> 2;
 }
 
-function byteLength(string, encoding) {
-  if (typeof string !== 'string') {
-    if (isArrayBufferView(string) || isAnyArrayBuffer(string)) {
-      return string.byteLength;
-    }
-
-    throw new ERR_INVALID_ARG_TYPE(
-      'string', ['string', 'Buffer', 'ArrayBuffer'], string
-    );
+const encodingOps = {
```

Comment:
Maybe add a comment that for performance reasons it's imperative that all the entries have the same shape / hidden map? (I.e., users of an ops object shouldn't go poly- or megamorphic.)